### DISCUSSION TITLE
Looking for platform pitch variable in StrucCtrl.f90

### DISCUSSION THREAD

**Discussion by relhamoud** [OpenFAST/openfast#2371] [Q&A] [answered]
Hello everyone,

I am trying to modify the code in the module _StrucCtrl.f90_  in OpenFAST aiming for active structural control. 
I am looking for the variable that gives platform pitch and i did not succeed to get it.
Could you help me please ?

Thank you in advance.

Best Regards,

**Comment by jjonkman** [comment, marked as answer]
Dear @relhamoud,

The source code of OpenFAST employs data encapsulation, so, modules can only access data known by another module through clearly defined module-level inputs and outputs.  I don't see that platform pitch is currently an input to the structural control (StC) submodel of ServoDyn.  That said, platform motion (including pitch) is an input to ServoDyn (via `u%PtfmMotionMesh`).  So, to change StC so that it knows platform pitch, you should also change ServoDyn and the StC registry file to pass platform motion (or just platform pitch, if that is all you want) from ServoDyn to StC.

Best regards,
